linux/libdriver/base/rollup-driver-api.h

```diff
@@ -0,0 +1,103 @@
+/** @file
+ * @defgroup rollup rollup
+ * Abstraction of the Cartesi rollup driver API
+ *
+ * Lets look at some code:
+ *
+ * @include examples/rollup.c
+ *
+ * @ingroup driver
+ * @{ */
+#ifndef ROLLUP_API_H
+#define ROLLUP_API_H
+#include<stdbool.h>
+#include<stddef.h>
+#include<stdint.h>
+
+enum {
+	ROLLUP_DRIVER_INVALID_STATE = -1, /**< @ref rollup_driver_loop invalid state */
+	ROLLUP_DRIVER_ADVANCE_STATE =  0, /**< @ref rollup_driver_loop advance state */
+	ROLLUP_DRIVER_INSPECT_STATE =  1, /**< @ref rollup_driver_loop inspect state */
+};
+
+/** contents of @ref rollup_driver_t are implementation specific.
+ * Define it in @p <rollup.h> */
+typedef struct rollup_driver rollup_driver_t;
+
+/** Open the rollup device and initialize the driver
+ *
+ * @param [in] me A uninitialized @ref rollup_driver_t state
+ * @returns
+ * - 0 on success
+ * - -1 on failure */
+int rollup_driver_init(struct rollup_driver *me);
+
+/** Release the driver resources and close the rollup device
+ *
+ * @param [in] me A sucessfuly initialized state by @ref rollup_driver_init
+ * @note usage of @p me after this call is a BUG and will cause undefined behaviour */
+void rollup_driver_fini(struct rollup_driver *me);
+
+/** Create a output, verifiable or not, with the first @p n bytes of @ref
+ * rollup_driver_t.tx
+ *
+ * @param [in] me A sucessfuly initialized state by @ref rollup_driver_init
+ * @param [in] verifiable @p true if this output should be verifiable
+ * @param [in] n size in bytes of the output, contents from @p tx buffer.
+ * @return
+ * - 0 on success
+ * - ENOBUFS in case the requested size is too large. User must split the
+ *   message in smaller chunks. */
+int rollup_driver_write_output(struct rollup_driver *me, bool verifiable, uint64_t n);
```

Comment:
I wouldn't use a bool. I would use an `enum` with named constants.

Reply:
Better yet, have two entrypoints just like the lower-level driver. Why not?

Reply:
I was trying to make the API more compact. But either way is fine with me.